java/api/src/main/java/org/ray/api/id/ObjectId.java

```diff
@@ -21,21 +21,17 @@ public static ObjectId fromByteBuffer(ByteBuffer bb) {
     return new ObjectId(byteBuffer2Bytes(bb));
   }
 
-  /**
-   * Generate a nil ObjectId.
-   */
-  private static ObjectId genNil() {
-    byte[] b = new byte[LENGTH];
-    Arrays.fill(b, (byte) 0xFF);
-    return new ObjectId(b);
-  }
-
   /**
    * Generate an ObjectId with random value.
    */
   public static ObjectId fromRandom() {
     byte[] b = new byte[LENGTH];
     new Random().nextBytes(b);
+    // Fill all bits of flags to zero to set the following flag values:
+    // Is task: false
+    // Object type: put
+    // Transport type: raylet
+    Arrays.fill(b, TaskId.LENGTH, TaskId.LENGTH + FLAGS_LENGTH, (byte) 0);
```

Comment:
Add a NOTE that this is tightly coupled with the ObjectID definition in C++, and if that changes this must be changed as well.

Reply:
ideally you can set the flags separately so that it won't break even if the value of the flags change, but it might be a bit overkill.

Reply:
How about now? Check my updated code.

Reply:
looks good to me. thanks